### PR TITLE
Fixed invalid "nano_" address handling in NanoUtil.addressToPublic

### PR DESCRIPTION
**Pull request by mkej**
`NanoUtil.addressToPublic` was only handling `xrb_`-prefixed addresses. This in turn caused one of the 45 `PreconfiguredRepresentatives` to be unusable - `nano_1thingspmippfngcrtk1ofd3uwftffnu4qu9xkauo9zkiuep6iknzci3jxa6`.

Functions handling this data do not fail, but return longer hex strings and byte arrays for public keys, which makes it very hard to observe and debug.

### CODE REVIEW

File: app/src/main/java/co/nano/nanowallet/NanoUtil.java [hunk 1]
```diff
@@ -250,8 +250,8 @@ public static String publicToAddress(String public_key) {
      * @return Public Key
      */
     public static String addressToPublic(String encoded_address) {
-        Sodium sodium = NaCl.sodium();
-        String data = encoded_address.substring(4, 56);
+        NaCl.sodium();
```
Comment (PlasmaPower) [inline app/src/main/java/co/nano/nanowallet/NanoUtil.java:253]: Why this change?

Comment (mkej) [reply app/src/main/java/co/nano/nanowallet/NanoUtil.java:253]: If you're asking about skipping the local `sodium` variable - it was not used, nor is required for Sodium to work. The `NaCl.sodium()` call can safely be done only once for the whole app, as it's responsible for loading native library and initializing it properly. Should either be done on application start, or in a `static` initializer of `NanoUtil`, no need to repeat it in every method.

Comment (PlasmaPower) [reply app/src/main/java/co/nano/nanowallet/NanoUtil.java:253]: Oh, I thought we were calling methods on this variable.
